compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -235,7 +235,28 @@ impl<'a> Parser<'a> {
         let span_lo = self.token.span;
         let (params, span) = if self.eat_lt() {
             let params = self.parse_generic_params()?;
-            self.expect_gt()?;
+
+            if let Some(last) = params.last()
+                && last.ident.as_str().to_ascii_lowercase() == kw::Const.as_str()
+            {
+                // We might have a typo'd `Const` that was parsed as a type parameter.
+                let span = last.ident.span;
+                let mut err = self.struct_span_err(
+                    span,
+                    format!("`const` keyword was mistyped as `{}`", last.ident.as_str()),
+                );
+                err.span_suggestion_verbose(
+                    span,
+                    "use the `const` keyword",
+                    kw::Const.as_str(),
+                    Applicability::MachineApplicable,
+                );
+                err.emit();
+                return Err(err);
+            } else {
+                self.expect_gt()?;
+            }
```

Comment:
Should this suggestion happen inside `parse_generic_params` or `parse_ty_param`?
Can it be made an actual recovery = if `self.may_recover()`, find a way to return `Ok` with a sensible AST?

Reply:
I *think* I got the recovery as `GenericParam` working

Reply:
Had to `attrs.clone()` a few places, not sure if I can avoid it